scripts/irods/test/test_all_rules.py

```diff
@@ -663,6 +663,82 @@ def test_msiServerMonPerf_default_3736(self):
 
         self.rods_session.assert_icommand("irule -F " + rule_file);
 
+    @unittest.skipUnless(plugin_name == 'irods_rule_engine_plugin-irods_rule_language', 'only applicable for irods_rule_language REP')
+    def test_msiAddKeyValToMspStr_works_with_empty_string_6918(self):
+        collection_to_free = os.path.join(self.admin.session_collection, 'freeMe')
+
+        # Generate trash
+        self.admin.assert_icommand(['itouch', collection_to_free])
+        self.admin.assert_icommand(['irm', collection_to_free])
+
+        rule_file="test_msiAddKeyValToMspStr.r"
+        rule_string= '''
+msiTestAddKeyValToMspStr {{
+    *Flag = "";
+    *Coll = "{trash_col}";
+    msiAddKeyValToMspStr("irodsAdminRmTrash", "", *Flag);
+    msiRmColl(*Coll, *Flag, *result);
```

Comment:
How does L680 help prove `msiAddKeyValToMspStr` handles empty strings correctly?

Reply:
Added test to check `msiAddKeyValToMspStr`, kept this one to test `msiRmColl`.